codec/decode_test.go

```diff
@@ -381,26 +379,22 @@ func TestDecodeFixedWidthInts(t *testing.T) {
 			t.Errorf("Fail: input %d got %d expected %d", test.val, output, test.output)
 		}
 	}
-}
 
-func TestDecodeInts(t *testing.T) {
-	for _, test := range decodeIntTests {
+	for _, test := range decodeFixedWidthIntTestsInt64 {
 		output, err := Decode(test.val, int64(0))
 		if err != nil {
 			t.Error(err)
-		} else if output != test.output {
```

Comment:
Please don't cast these, tests should work without

Reply:
it doesn't work because Decode returns an interface with underlying type int, I can cast `output` back to the same type as `test.output` though and not cast `test.output`? eg. `uint64(output.(int)) != test.output` that seems to work ok

Reply:
I fail to see why these worked before but now don't

Reply:
I think the diffs aren't comparing the right lines? if you look at the tests it seems to be different tests

Reply:
> I fail to see why these worked before but now don't

I think my comment above explains why?

Reply:
we're comparing different tests I think, I didn't actually change this for this PR, I just got rid of the tests for compact ints using int64 since they're encoded as LE now and added them to the DecodeFixedWidthInt tests which were already doing the `output.(int)` thing